lib/measures/urban_geometry_creation/measure.rb

```diff
@@ -789,45 +165,44 @@ def run(model, runner, user_arguments)
     end
 
     @origin_lat_lon = OpenStudio::PointLatLon.new(min_lat, min_lon, 0)
-    
+
     site = model.getSite
     site.setLatitude(@origin_lat_lon.lat)
     site.setLongitude(@origin_lat_lon.lon)
-    
-    if feature[:properties][:surface_elevation]
-      surface_elevation = feature[:properties][:surface_elevation].to_f
+
+    if feature.get(:surface_elevation)
+      surface_elevation = feature.get(:surface_elevation).to_f
       surface_elevation = OpenStudio::convert(surface_elevation, 'ft', 'm').get
       site.setElevation(surface_elevation)
     end
-    
-    feature_type = feature[:properties][:type]
-    
+
+    feature_type = feature.get(:type)
+
     if feature_type == 'Building'
-    
       # make requested building
-      spaces = create_building(feature, :space_per_floor, model)
+      spaces = URBANopt::GeoJSON::BuildingCreation.create_building(feature, :space_per_floor, model, @origin_lat_lon, @runner)
       if spaces.nil? 
         @runner.registerError("Failed to create spaces for building '#{name}'")
         return false
       end
       
       # DLM: temp hack
-      building_type = feature[:properties][:building_type]
+      building_type = feature.get(:building_type)
       if building_type == 'Vacant'
         max_z = 0
         spaces.each do |space|
           bb = space.boundingBox
           max_z = [max_z, bb.maxZ.get].max
         end
-        shading_surfaces = create_photovoltaics(feature, max_z + 1, model)
+        shading_surfaces = URBANopt::GeoJSON::Helper.create_photovoltaics(feature, max_z + 1, model, @origin_lat_lon, @runner)
       end
       
       # make other buildings to convert to shading
       convert_to_shades = []
       if surrounding_buildings == "None"
         # no-op
       else
-        convert_to_shades = create_other_buildings(feature, surrounding_buildings, model)
+        convert_to_shades = geojson_gem.create_other_buildings(feature, surrounding_buildings, model, @origin_lat_lon, @runner)
```

Comment:
Is this method planned to move?

Reply:
I do have a question about this method! So the measure test (in `./lib/measures/urban_geometry_creation/tests/urban_geometry_creation_test.rb`) doesn't test a building with surrounding buildings.  And methods like create_other_buildings only get called when there are surrounding buildings. Should I add another measure test that tests this scenario?

Reply:
Yes we should add a test for this scenario.  I'm not sure this method is the right way to do things though.  The methods called for the main footprint of interest (referenced by id) generate a full model.  This function creates shading geometry only, it doesn't have equipment, loads, HVAC, etc.  So I think we might rather have two methods:

`create_building(feature, model)` and `create_shading(feature, model)`. 

Then you could have another method `find_surrounding_features` and do something like this:

```
main_feature = geo_file.get_feature(id)
main_building = BuildingCreation.create_building(main_feature, model)
surrounding_features = main_feature.find_surrounding_features
surrounding_features.each {|feature| BuildingCreation.create_shading(feature, model)
```